tests/fermi/test_parity_mapping.py

```diff
@@ -0,0 +1,769 @@
+
+    # converting to Pauli representation for comparison because
+    # qml.equal isn't playing nicely with term ordering
+    assert pauli_sentence(op) == pauli_sentence(result)
+
+
+@pytest.mark.parametrize("wire_map, ops", WIRE_MAP_FOR_FERMI_SENTENCE)
+def test_providing_wire_map_fermi_sentence_to_ps(wire_map, ops):
+    fs = FermiSentence(
+        {FermiWord({(0, 0): "+", (1, 1): "-"}): 1, FermiWord({(0, 0): "+", (1, 0): "-"}): 1}
+    )
+    n_qubits = 4
+    op = parity_transform(fs, n_qubits, wire_map=wire_map, ps=True)
+    result_op = qml.sum(*ops)
+    ps = pauli_sentence(result_op)
+
+    ps.simplify()
+    op.simplify()
+
+    assert ps == op
+
+
+WIRE_MAP_FOR_FERMI_WORDS = [
+    (
+        None,
+        [
+            qml.s_prod(-0.25j, qml.PauliY(0)),
+            qml.s_prod(0.25 + 0j, qml.prod(qml.PauliX(0), qml.PauliZ(1))),
+            qml.s_prod(0.25 + 0j, qml.PauliX(0)),
+            qml.s_prod(-0.25j, qml.prod(qml.PauliY(0), qml.PauliZ(1))),
+        ],
+    ),
+    (
+        {0: 3, 1: 2},
+        [
+            qml.s_prod(-0.25j, qml.PauliY(3)),
+            qml.s_prod(0.25 + 0j, qml.prod(qml.PauliX(3), qml.PauliZ(2))),
+            qml.s_prod(0.25 + 0j, qml.PauliX(3)),
+            qml.s_prod(-0.25j, qml.prod(qml.PauliY(3), qml.PauliZ(2))),
+        ],
+    ),
+    (
+        {0: "b", 1: "a"},
+        [
+            qml.s_prod(-0.25j, qml.PauliY("b")),
+            qml.s_prod(0.25 + 0j, qml.prod(qml.PauliX("b"), qml.PauliZ("a"))),
+            qml.s_prod(0.25 + 0j, qml.PauliX("b")),
+            qml.s_prod(-0.25j, qml.prod(qml.PauliY("b"), qml.PauliZ("a"))),
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("wire_map, ops", WIRE_MAP_FOR_FERMI_WORDS)
+def test_providing_wire_map_fermi_word_to_operation(wire_map, ops):
+    w = FermiWord({(0, 0): "+", (1, 1): "+"})
+    n_qubits = 4
+    op = parity_transform(w, n_qubits, wire_map=wire_map)
+    result = qml.sum(*ops)
+
+    op.simplify()
+
+    # converting to Pauli representation for comparison because
+    # qml.equal isn't playing nicely with term ordering
+    assert pauli_sentence(op) == pauli_sentence(result)
+
+
+@pytest.mark.parametrize("wire_map, ops", WIRE_MAP_FOR_FERMI_WORDS)
+def test_providing_wire_map_fermi_word_to_ps(wire_map, ops):
+    w = FermiWord({(0, 0): "+", (1, 1): "+"})
+    n_qubits = 4
+    op = parity_transform(w, n_qubits, wire_map=wire_map, ps=True)
+    result_op = qml.sum(*ops)
+    ps = pauli_sentence(result_op)
+
+    ps.simplify()
+    op.simplify()
+
+    assert ps == op
+
+
+fs1 = FermiSentence({fw1: 1})
+
+
+@pytest.mark.parametrize(
+    "fermi_op, qubit_op_data, tol",
+    (
+        (fw1, (-0.25j, (0.25 + 0j), (0.25 + 0j), 0.25j), None),
+        (fw1, (-0.25j, 0.25, 0.25, 0.25j), 0.0),
+        (fw1, (-0.25j, 0.25, 0.25, 0.25j), 1.0e-12),
+        (fs1, (-0.25j, (0.25 + 0j), (0.25 + 0j), 0.25j), None),
+        (fs1, (-0.25j, 0.25, 0.25, 0.25j), 0.0),
+        (fs1, (-0.25j, 0.25, 0.25, 0.25j), 1.0e-12),
+    ),
```

Comment:
I think we should test if a non-zero but still negligible imaginary component gets mapped to 0. One way to do this would be to set the tolerance to 0.3. The expected result would be the first and last amplitudes get mapped to 0.

Reply:
I have added one each for fermiword and fermisentence.